sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -46,6 +46,21 @@ trait CheckAnalysis extends PredicateHelper {
     }).length > 1
   }
 
+  private def checkLimitClause(limitExpr: Expression): Unit = {
+    if (!limitExpr.foldable) {
+      failAnalysis(
+        "The argument to the LIMIT clause must evaluate to a constant value. " +
+        s"Limit:${limitExpr.sql}")
+    }
+    limitExpr.eval() match {
+      case o: Int if o >= 0 => // OK
+      case o: Int => failAnalysis(
+        s"number_rows in limit clause must be equal to or greater than 0. number_rows:$o")
+      case o => failAnalysis(
```

Comment:
You know, we can do it, but we need to fix the other parts. Let me try it. Thanks!